src/udev/udevd.c

```diff
@@ -1541,7 +1559,9 @@ static int on_sigchld(sd_event_source *s, const struct signalfd_siginfo *si, voi
                         device_tag_index(worker->event->dev, NULL, false);
 
                         /* Forward kernel event to libudev listeners */
-                        device_broadcast(manager->monitor, worker->event->dev);
+                        device_broadcast(manager->monitor, worker->event->dev,
+                                         WIFEXITED(status) ? WEXITSTATUS(status) :
```

Comment:
doesn't matter but may write this as `EVENT_RESULT_EXIT_STATUS_BASE + WEXITSTATUS(status)`? just to make things even more explicit